antarest/study/storage/variantstudy/business/utils_binding_constraint.py

```diff
@@ -53,7 +53,7 @@ def apply_binding_constraint(
             # Cluster IDs are stored in lower case in the binding constraints file.
             area, cluster_id = link_or_cluster.split(".")
             thermal_ids = {thermal.id.lower() for thermal in study_data.config.areas[area].thermals}
```

Comment:
On devrait avoir une variable locale en dehors de la boucle `for` pour stocker la liste des zones, cela pourrait être plus efficace et plus maintenable.

```python
areas = study_data.config.areas
for link_or_cluster in coeffs:
    if "%" in link_or_cluster:
        ...
```